src/lifetimes.rs

```diff
@@ -61,6 +64,7 @@ fn check_fn_inner(cx: &LateContext, decl: &FnDecl, slf: Option<&ExplicitSelf>,
         span_lint(cx, NEEDLESS_LIFETIMES, span,
                   "explicit lifetimes given in parameter types where they could be elided");
     }
+    report_extra_lifetimes(cx, decl, &generics.lifetimes);
```

Comment:
It only reports lifetimes defined in the function lt param list that aren't used in the argument/return list, not the other way around. Adding a test.

(forgot to `git add` the tests folder too, did that now)